src/store/model/amazon-uk.ts

```diff
@@ -0,0 +1,95 @@
+import {Link, Store} from './store';
+import {logger} from '../../logger';
+import {parseCard} from './helpers/card';
+
+export const AmazonUk: Store = {
+	backoffStatusCodes: [403, 429, 503],
+	labels: {
+		captcha: {
+			container: 'body',
+			text: ['enter the characters you see below']
+		},
+		inStock: {
+			container: '#availability',
+			text: ['in stock']
+		},
+		maxPrice: {
+			container: 'span[class*="PriceString"]'
+		},
+		outOfStock: [
+			{
+				container: '#availability',
+				text: ['out of stock', 'unavailable']
+			},
+			{
+				container: '#backInStock',
+				text: ['unavailable']
+			}
+		]
+	},
+	links: [
+		{
+			brand: 'test:brand',
+			cartUrl: 'https://www.amazon.co.uk/gp/aws/cart/add.html?ASIN.1=B081265T5Z&Quantity.1=1',
+			model: 'test:model',
+			series: 'test:series',
+			url: 'https://www.amazon.co.uk/dp/B081265T5Z/'
+		}
+	],
+	linksBuilder: {
```

Comment:
I imagine we could use this for the other Amazon stores, but could be very wrong there.

Reply:
I imagine the `builder` function would work between stores, but determining the `urls[].url[]` would require a manual map per region. At a minimum, the following may work (not tested it compiles let alone works):

```ts
function getAmazonQueryString(search: string, categoryId: number, includeOutOfStockId: number) {
    return [
        `k=${encodeURIComponent(search)}`,
        'rh=' + encodeURIComponent([
            `n:${categoryId}`,
            `p_n_availability:${includeOutOfStockId}`
        ].join(','))
    ].join('&');
}

const queryString = getAmazonQueryString('+"RTX 3080" -2080 -GTX', 430500031, 419162031);
const domain = 'www.amazon.co.uk';
const url = `https://${domain}/s?${queryString}`;
```

Reply:
That makes sense to me. Maybe we can bring this in at a later point. Thanks for checking this out.